Description: don't try to parse empty sections
 Since smartmontools 6.0, the output is divided in sections regardless of the
 drive having or not SMART capabilities; in the latter case the last section
 is empty. In that case, just skip it (parsing empty sections causes a crash).
Author: Francesco Presel <f.presel@alice.it>
Bug-Debian: http://bugs.debian.org/722989
Last-Update: <2014-01-23>

--- a/src/applib/smartctl_parser.cpp
+++ b/src/applib/smartctl_parser.cpp
@@ -202,7 +202,10 @@
 			&& (section_start_pos = s.find("===", section_start_pos)) != std::string::npos) {
 
 		tmp_pos = s.find("\n", section_start_pos);  // works with \r\n too.
-
+		if (tmp_pos == std::string::npos) { // empty section: skip
+			break;
+		}
+			
 		// trim is needed to remove potential \r in the end
 		std::string section_header = hz::string_trim_copy(s.substr(section_start_pos,
 				(tmp_pos == std::string::npos ? tmp_pos : (tmp_pos - section_start_pos)) ));
